alibi_detect/utils/saving.py

```diff
@@ -948,856 +71,384 @@ def load_detector(filepath: Union[str, os.PathLike], **kwargs) -> Data:
+    if cfg is not None:
+        cfg = cfg()  # TODO - can just do detector.get_config() once all Detectors have a .get_config()
+    else:
+        raise ValueError('{} is not supported by `save_detector`.'.format(detector_name))
 
-    Parameters
-    ----------
+    # Save x_ref
+    save_path = filepath.joinpath('x_ref.npy')
+    np.save(str(save_path), cfg['x_ref'])
+    cfg.update({'x_ref': 'x_ref.npy'})
+
+    # Save preprocess_fn
+    preprocess_fn = cfg.get('preprocess_fn', None)
+    if preprocess_fn is not None:
+        logger.info('Saving the preprocess_fn function.')
+        preprocess_cfg = _save_preprocess(preprocess_fn, backend, cfg['input_shape'], filepath)
+        cfg['preprocess_fn'] = preprocess_cfg
+
+    # Serialize kernel
+    kernel = cfg.get('kernel', None)
+    if kernel is not None:
+        device = getattr(detector, 'device', None)
+        device = device.type if device is not None else None
+        cfg['kernel'] = _save_kernel(kernel, filepath, device=device)
+        if isinstance(kernel, dict):  # serialise proj from DeepKernel
+            cfg['kernel']['proj'], _ = _save_model_config(kernel['proj'], base_path=filepath,
+                                                          input_shape=cfg['input_shape'], backend=backend)
+
+    # ClassifierDrift and SpotTheDiffDrift specific artefacts.
+    # Serialize detector model
+    model = cfg.get('model', None)
+    if model is not None:
+        model_cfg, _ = _save_model_config(model, base_path=filepath, input_shape=cfg['input_shape'], backend=backend)
+        cfg['model'] = model_cfg
+
+    # Serialize dataset
+    dataset = cfg.get('dataset', None)
+    if dataset is not None:
+        dataset_cfg, dataset_kwargs = _serialize_function(dataset, filepath, Path('dataset'))
+        cfg.update({'dataset': dataset_cfg})
+        if len(dataset_kwargs) != 0:
+            cfg['dataset']['kwargs'] = dataset_kwargs
+
+    # Serialize reg_loss_fn
+    reg_loss_fn = cfg.get('reg_loss_fn', None)
+    if reg_loss_fn is not None:
+        reg_loss_fn_cfg, _ = _serialize_function(reg_loss_fn, filepath, Path('reg_loss_fn'))
+        cfg['reg_loss_fn'] = reg_loss_fn_cfg
+
+    # Save initial_diffs
+    initial_diffs = cfg.get('initial_diffs', None)
+    if initial_diffs is not None:
+        save_path = filepath.joinpath('initial_diffs.npy')
+        np.save(str(save_path), initial_diffs)
+        cfg.update({'initial_diffs': 'initial_diffs.npy'})
+
+    # Save config
+    write_config(cfg, filepath)
+
+
+def write_config(cfg: dict, filepath: Union[str, os.PathLike]) -> dict:
+    """
+    Save an unresolved detector config dict to a TOML file.
+
+    Parameters
+    ----------
+    cfg
+        Unresolved detector config dict.
     filepath
-        Saved model directory.
-    state_dict
-        Dictionary containing the `n_gmm`, `latent_dim` and `recon_features` parameters.
+        Filepath to directory to save 'config.toml' file in.
 
     Returns
```

Comment:
I just did a `grep` and it seems the return value is not used anywhere, do we need to keep it?

Reply:
Removed the return. Thanks!